src/main/java/com/ledikom/service/BotService.java

```diff
@@ -66,6 +67,19 @@ public void processCouponsInMap() {
         CouponService.userCoupons.entrySet().removeIf(userCoupon -> userCoupon.getValue().getExpiryTimestamp() < System.currentTimeMillis() - 5000);
     }
 
+    @Scheduled(cron = "0 0 0 * * *")
+    public void deleteCouponsIfExpired() {
+        List<Coupon> coupons = couponService.getAllCoupons();
+
+        LocalDateTime currentDateTime = LocalDateTime.now();
```

Comment:
Use moscow time zone here either, to compare moscow time with moscow time, because it won't be "after" in the below if statement